de-lib/src/main/java/org/iplantc/de/client/models/UserInfo.java

```diff
@@ -73,15 +77,19 @@ public void init(String userInfoJson) {
         dataInfo = userBootstrap.getDataInfo();
         preferences = userBootstrap.getPreferences();
         session = userBootstrap.getSession();
-        workspace = userBootstrap.getWorkspace();
+        appsInfo = userBootstrap.getAppsInfo();
+        workspace = appsInfo.getWorkspace();
+        systemIds = appsInfo.getSystemsIds();
     }
 
     public boolean hasErrors() {
         return hasUserProfileError() ||
                hasDataInfoError() ||
                hasPreferencesError() ||
                hasSessionError() ||
-               hasWorkspaceError();
+               hasWorkspaceError() ||
+               hasSystemsError() ||
+               hasAppsInfoError();
```

Comment:
Actually, there wouldn't be separate errors for the `workspace` or `system_ids` keys. These come from the apps service, so if apps is down, then only the parent `apps_info` key will contain an error message.
So `hasAppsInfoError` should cover all 3 of these.